platform/android/java/lib/src/org/godotengine/godot/io/file/MediaStoreData.kt

```diff
@@ -230,6 +232,75 @@ internal class MediaStoreData(context: Context, filePath: String, accessFlag: Fi
 			)
 			return updated > 0
 		}
+
+		fun getUriFromDirectoryPath(context: Context, directoryPath: String): Uri? {
+			if (!directoryExists(directoryPath)) {
+				return null
+			}
+			// Check if the path is under external storage
+			val externalStorageRoot = Environment.getExternalStorageDirectory().absolutePath
+			if (directoryPath.startsWith(externalStorageRoot)) {
+				val relativePath = directoryPath.replaceFirst(externalStorageRoot, "").trim('/')
+				val uri = Uri.Builder()
+					.scheme("content")
+					.authority("com.android.externalstorage.documents")
+					.appendPath("document")
+					.appendPath("primary:$relativePath")
+					.build()
+				return uri
+			}
+			return null
+		}
+
+		fun getFilePathFromUri(context: Context, uri: Uri): String? {
+			// Converts content uri to filepath
+			val id = getIdFromUri(uri) ?: return null
+
+			if (uri.authority == "com.android.externalstorage.documents") {
+				val split = id.split(":")
+				val fileName = split.last()
+				val relativePath = split.dropLast(1).joinToString("/")
+				val fullPath = File(Environment.getExternalStorageDirectory(), "$relativePath/$fileName").absolutePath
+				return fullPath
+			} else {
+				val id = id.toLongOrNull() ?: return null
+				val dataItems = queryById(context, id)
+				return if (dataItems.isNotEmpty()) {
+					val dataItem = dataItems[0]
+					File(Environment.getExternalStorageDirectory(), File(dataItem.relativePath, dataItem.displayName).toString()).absolutePath
+				} else {
+					null
+				}
+			}
+		}
+
+		private fun getIdFromUri(uri: Uri): String? {
+			return try {
+				if (uri.authority == "com.android.externalstorage.documents") {
+					val documentId = uri.lastPathSegment ?: throw NumberFormatException("Invalid URI: $uri")
```

Comment:
Should this be an `IllegalArgumentException` instead?

Reply:
I initially used `NumberFormatException` because that was the exception thrown by Android. However, I now believe that `IllegalArgumentException` might be more appropriate in this context.